text/0141-linking-traces.md

```diff
@@ -0,0 +1,367 @@
+
+Span links are supported by all OpenTelemetry platforms we currently use (or plan on using) for our SDKs: [JavaScript (Node)](https://opentelemetry.io/docs/languages/js/instrumentation/#span-links), [Python](https://opentelemetry.io/docs/languages/python/instrumentation/#adding-links), [Elixir](https://opentelemetry.io/docs/languages/erlang/instrumentation/#linking-the-new-span) and [Java](https://opentelemetry.io/docs/languages/java/api-components/#span).
+
+
+In an OTLP span export, span links are serialized as follows:
+
+```json
+// output was shortened to important fields
+{
+  "resourceSpans": [
+    {
+      "scopeSpans": [
+        {
+          "spans": [
+            {
+              "traceId": "627a2885119dcc8184fae7eef09438cb",
+              "spanId": "ec53f20e4318380d",
+              "links": [ // <-- link
+                {
+                  "attributes": [ // <-- link attributes
+                    {
+                      "key": "sentry.link.type",
+                      "value": {
+                        "stringValue": "parent"
+                      }
+                    }
+                  ],
+                  // span and trace linked to (from span.spanContext()): 
+                  "spanId": "6c71fc6b09b8b716", 
+                  "traceId": "627a2885119dcc8184fae7eef09438cb", 
+                }
+              ],
+              "droppedLinksCount": 0
+            },
+            {
+              "traceId": "627a2885119dcc8184fae7eef09438cb",
+              "spanId": "6c71fc6b09b8b716",
+              "parentSpanId": "13d4cb0933154c29",
+              "name": "span-links-txn",
+              "links": [],
+              "droppedLinksCount": 0
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+
+## Current Session Models in Sentry SDKs
+
+Sentry SDKs do actually send session data, in fact even two types of sessions. However, neither of the two sessions are used and associated with a trace. 
+
+- SDK error sessions: SDKs currently send a session that counts and describes the error status of such a session. Depending on the error, the session is marked as crashed, abnormal or healthy, which is information that powers our Release (Health) product. These sessions are fundamentally flawed though because (at least in Browser JS) they only last as long as the currently loaded page. Every soft or hard navigation causes the session to be ended and a new one to be started. 
+- Replay Session: Frontend SDKs persist a replayId in the browser's `sessionStorage` which (even though the name does not suggest it) more accurately models a session than the SDKs' error sessions. We cannot have a hard dependency on this session as Replay is an extra product by Sentry, meaning this replayId is not always reliably set. However, the model of persisting the replay id can be used as a blueprint for how we could persist the last traceId. 
```

Comment:
I think we should align on what a session is at Sentry. Having different types of sessions for Session Replay and the rest of the product will not only confuse us but also our users. If we plan on using the sessionStorage as a blueprint, we should instead consider extracting the logic from Session Replay.

Reply:
I agree 100% and fwiw, we've started talking about making the replay session the main (and only) session in the SDK.

Reply:
Here's a link to the tracking issue: https://github.com/getsentry/sentry-javascript/issues/14478